pkg/scheduler/statusupdater.go

```diff
@@ -0,0 +1,131 @@
+package scheduler
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/kyma-incubator/reconciler/pkg/cluster"
+	"github.com/kyma-incubator/reconciler/pkg/keb"
+	"github.com/kyma-incubator/reconciler/pkg/model"
+	"go.uber.org/zap"
+)
+
+const (
+	CHANNEL_SIZE    = 100
+	progressTimeout = 20 * time.Minute
+)
+
+type ClusterStatusUpdater struct {
+	inventory     cluster.Inventory
+	clusterState  cluster.State
+	updateChannel chan Update
+	statusMap     map[string]string
+	logger        *zap.SugaredLogger
+}
+
+type Update struct {
+	component      string
+	operationState string
+}
+
+func NewClusterStatusUpdater(inventory *cluster.Inventory, clusterState cluster.State, components []*keb.Components, logger *zap.SugaredLogger) ClusterStatusUpdater {
+	statusUpdater := ClusterStatusUpdater{inventory: *inventory, clusterState: clusterState, logger: logger}
+	statusUpdater.statusMap = make(map[string]string)
+	for _, comp := range components {
+		statusUpdater.statusMap[comp.Component] = StateInProgress
+	}
+	statusUpdater.reconciling()
+	statusUpdater.updateChannel = make(chan Update, CHANNEL_SIZE)
+	return statusUpdater
+}
+
+func (su *ClusterStatusUpdater) Run() {
+	timeout := time.After(progressTimeout)
+	for {
+		select {
+		case update := <-su.updateChannel:
+			su.statusMap[update.component] = update.operationState
+			if update.operationState == StateInProgress {
+				su.reconciling()
```

Comment:
To save DB requests, it would be better to change the reconciling state of the cluster only at the beginning to `reconcile`. 

In this loop it would be enough to check only for DONE or ERROR updates.